msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/MessageForumsForumManagerImpl.java

```diff
@@ -1571,4 +1573,28 @@ public boolean doesRoleHavePermissionInTopic(final Long topicId, final String ro
 			Number countRows = getHibernateTemplate().execute(hcb);
 			return countRows.intValue() > 0;
 		}
+		
+		public String getAllowedGroupForRestrictedTopic(final Long topicId, final String permissionName) {
+			if (topicId == null) {
+				throw new IllegalArgumentException("Null Argument");
+			}
+			HibernateCallback<String> hcb = session -> (String) session
+				.getNamedQuery("findAllowedGroupInTopic")
+				.setLong("id", topicId)
+				.setString("permissionLevelName", permissionName)
+				.uniqueResult();
+			return getHibernateTemplate().execute(hcb);
```

Comment:
I would like to understand why the unique result was expected here! CC @ern

Reply:
If I remember well, this feature would create automatically one topic/forum for each selected group, following the pattern TOPIC TITLE - GROUP NAME X. So the unicity was probably for that reason.

I think this behaviour was copied from somewhere where it was already like that, it wasn't intending to provide a full group visibility feature. I can see why in the long term this might have been causing confusion though...